Jenkinsfile.tag

```diff
@@ -0,0 +1,42 @@
+pipeline {
+    agent { label 'upbound-gce' }
+
+    parameters {
+        string(name: 'version', defaultValue: '', description: 'The version you are releasing, for example v0.4.0')
+        string(name: 'commit', defaultValue: '', description: 'Optional commit hash for this release, for example 56b65dba917e50132b0a540ae6ff4c5bbfda2db6. If empty the latest commit hash will be used.')
+    }
+
+    options {
+        disableConcurrentBuilds()
+        timestamps()
+    }
+
+    environment {
+        GITHUB_UPBOUND_BOT = credentials('github-upbound-jenkins')
+    }
+
+    stages {
+
+        stage('Prepare') {
+            steps {
+                 // github credentials are not setup to push over https in jenkins. add the github token to the url
+                sh "git config remote.origin.url https://${GITHUB_UPBOUND_BOT_USR}:${GITHUB_UPBOUND_BOT_PSW}@\$(git config --get remote.origin.url | sed -e 's/https:\\/\\///')"
```

Comment:
Perhaps in all of the pipeline that take this approach, we could simplify the authentication requirements by using `git config --global --add url."git@github.com:".insteadOf "https://github.com/"` and then relying on ssh credentials (which I imagine are already available or need less configuration).

Reply:
Yeah - I'm thinking it'd be beneficial to share code between the different pipelines first, so that any upgrades we do would apply for all of them